pulseguardian/guardian.py

```diff
@@ -219,22 +230,66 @@ def back_to_normal_email(self, user, queue_data):
            queue_data['messages'], self.del_queue_size)
 
         if self.emails and user.email is not None:
-            sendemail(subject=subject, from_addr=config.email_from,
-                      to_addrs=[user.email], username=config.email_account,
-                      password=config.email_password, text_data=body,
-                      server=config.email_smtp_server,
-                      port=config.email_smtp_port,
-                      use_ssl=config.email_ssl)
+            self._sendemail(
+                subject=subject, to_addrs=[user.email], text_data=body)
+
+    def notify_connection_error(self):
+        if self.emails and not self._connection_error_notified:
+            ex_details = traceback.format_exc()
+            admin_emails = [user.email 
+                for user in User.query.filter_by(admin=True)]
+            subject = "Pulse error: Can't connect to Pulse"
+            body = '''Could not connect to Pulse.
+Details:
+Rabbit Url: {0}
+Rabbit User: {1}
+Error:
+{2}
+'''.format(config.rabbit_management_url, config.rabbit_user, ex_details)
+
+            self._sendemail(
+                subject=subject, to_addrs=admin_emails, text_data=body)
+            logging.error(subject + body)
+            self._connection_error_notified = True
+
+    def notify_unknown_error(self):
+        if self.emails and not self._unknown_error_notified:
+            ex_details = traceback.format_exc()
+            admin_emails = [user.email 
+                for user in User.query.filter_by(admin=True)]
+            subject = "Pulse error: Unknown error"
+            body = '''Unknown error occured.
+Details:
+{0}
+'''.format(ex_details)
+            self._sendemail(
+                subject=subject, to_addrs=admin_emails, text_data=body)
+            logging.error(subject + body)
+            self._unknown_error_notified = True
 
     def guard(self):
         logging.info("PulseGuardian started")
-        while True:
-            queues = self.api.queues()
-            if queues:
-                self.monitor_queues(queues)
-            self.clear_deleted_queues(queues)
-            time.sleep(config.polling_interval)
 
+        while True:
+            try:
+                queues = self.api.queues()
+                if queues:
+                    self.monitor_queues(queues)
+                self.clear_deleted_queues(queues)
+
+                if (self._connection_error_notified or 
+                                    self._unknown_error_notified):
```

Comment:
Too much indentation. :) The standard I like is, for "if (" statements like this, just indenting another four spaces.  So like this:

if (self._connection_error_notified or 
        self._unknown_error_notified):